scripts/encapp_tests.py

```diff
@@ -0,0 +1,390 @@
+#!/usr/local/bin/ python3
+import os
+import subprocess
+import json
+import sys
+import argparse
+import re
+from datetime import datetime
+from plot_rd import RDPlot
+
+KEY_NAME_DISCRIPTION="description"
+KEY_NAME_INPUT_FILES="input_files"
+KEY_NAME_CODECS="codecs"
+KEY_NAME_ENCODE_RESOLUTIONS="encode_resolutions"
+KEY_NAME_RC_MODES="rc_modes"
+KEY_NAME_BITRATES="bitrates"
+KEY_NAME_I_INTERVALS="i_intervals"
+KEY_NAME_DURATION="duration"
+KEY_NAME_USE_SURFACE_ENC="use_surface_enc"
+
+
+
+sample_config_json_data = [
+    {
+        KEY_NAME_DISCRIPTION: "sample",
+        KEY_NAME_INPUT_FILES: [""],
+        KEY_NAME_USE_SURFACE_ENC: 1,
+        KEY_NAME_CODECS: ["hevc"],
+        KEY_NAME_ENCODE_RESOLUTIONS: ["1280x720"],
+        KEY_NAME_RC_MODES: ["cbr"],
+        KEY_NAME_BITRATES: [500,1000,1500,2000,2500],
+        KEY_NAME_I_INTERVALS: [2,10,60],
+        KEY_NAME_DURATION: 10
+    }
+]
+TEST_CLASS_NAME="com.facebook.encapp.CodecValidationInstrumentedTest"
+JUNIT_RUNNER_NAME="com.facebook.encapp.test/android.support.test.runner.AndroidJUnitRunner"
+ENCAPP_OUTPUT_FILE_NAME_RE = '.*_[0-9]+fps_[0-9]+x[0-9]+_[0-9]+bps_iint[0-9]+_m[0-9]+(?:\.mp4$|\.webm)'
+RD_RESULT_FILE_NAME = "rd_results.json"
+
+def run_cmd(cmd):
+    try:
+        print(cmd, sep=' ')
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        # print ('standard output: {}'.format(stdout.decode()))
+        # print ('standard error: {}'.format(stderr.decode()))
+    except:
+        raise Exception('Failed to run adb command')
+
+    return stdout.decode(), stderr.decode()
+
+def check_device(serial_no):
+    #check for devices
+    adb_cmd = 'adb devices -l'
+    stdout, stderr = run_cmd(adb_cmd)
+    device_serials = []
+
+    for line in stdout.split('\n'):
+        if line == "List of devices attached" or line=='':
+            continue
+        serial = line.split()[0]
+        device_serials.append(serial)
+
+    if len(device_serials) == 0:
+        print ("No device connected")
+        exit(0)
+    elif len(device_serials) > 1:
+        if (serial_no is None) :
+            print ("More than one devices connected. Please specifiy a device serial number")
+            exit(0)
+        elif (serial_no not in device_serials):
+            print ('Specified serial number "{}" is invalid.'.format(serial_no) )
```

Comment:
you should be coherent with using either " or ' for strings. FB's recommendation IIRC is single quotes